src/graphics/reproject-texture.js

```diff
@@ -332,72 +339,46 @@ class SimpleCache {
         }
         return this.map.get(key);
     }
-
-    clear() {
-        this.map.clear();
-    }
 }
 
-// per-device cache
-class DeviceCache {
-    constructor() {
-        this.cache = new SimpleCache();
-    }
+// cache, used to store:
+// - samples. we store these separately from textures since multiple devices can use the same
+//   set of samples.
+// - texures inside DeviceResourceCache.reprojectTextureCache - those are per device
+const samplesCache = new SimpleCache();
 
-    // get the cache entry for the given device and key
-    // if entry doesn't exist, missFunc will be invoked to create it
-    get(device, key, missFunc) {
-        return this.cache.get(device, () => {
-            const cache = new SimpleCache();
-            device.on('destroy', () => {
-                cache.map.forEach((value, key) => {
-                    value.destroy();
-                });
-                this.cache.map.delete(device);
-            });
-            return cache;
-        }).get(key, missFunc);
+const getCachedTexture = (device, key, getSamplesFnc) => {
```

Comment:
This function accesses `cache.reprojectTextureCache.map` three times (internals of `SimpleCache`), it should probably be a method of `SimpleCache` itself then, which will also shorten the code while using less object property lookups

Reply:
done, thanks!